File: stack.go
package main

import (
  "fmt"
  "strconv"
)

func get(top int, data [3]string) {
  fmt.Println("\n", top, data[0:top])
}

func push(top int, data [3]string, maximumSize int, newItem string) (int, [3]string) {
  if top < maximumSize {
    data[top] = newItem
    top++
    get(top, data)
  } else {
    fmt.Println("\n Stack ("+strconv.Itoa(top)+") is full, can't add", newItem)
  }
  return top, data
}

func pop(top int, data [3]string) int {
  if top != 0 {
    top--
    get(top, data)
  }
  return top
}

func main() {

  var top int = 0
  var data = [3]string{}
  var maximumSize int = 3

  top, data = push(top, data, maximumSize, "Kevin")

  top, data = push(top, data, maximumSize, "Sally")

  top, data = push(top, data, maximumSize, "David")

  top, data = push(top, data, maximumSize, "Sonya")

  top = pop(top, data)

  top, data = push(top, data, maximumSize, "Jessy")

  top, data = push(top, data, maximumSize, "Harry")

}

File: stack.java
class StackClass {
  int top = 0;
  String[] data = { "", "", "" };
  int maximumSize = 3;

  public void get() {
    System.out.print("\n " + this.top + " [ ");
    for (int i = 0; i < this.top; i++) {
      System.out.print(this.data[i] + " ");
    }
    System.out.print("]\n");
  }

  public void push(String newItem) {
    if (this.top < this.maximumSize) {
      this.data[this.top] = newItem;
      this.top++;
      this.get();
    } else {
      System.out.println("\n Stack (" + this.top + ") is full, can't add " + newItem);
    }
  }

  public void pop() {
    if (this.top != 0) {
      this.top--;
      this.get();
    }
  }
}

public class stack {
  public static void main(String[] args) {

    StackClass array = new StackClass();

    array.push("Kevin");

    array.push("Sally");

    array.push("David");

    array.push("Sonya");

    array.pop();

    array.push("Jessy");

    array.push("Harry");

  }
}
